### PR TITLE
Sanitize clipboard paste

### PR DESCRIPTION
**Pull request by Aussiemon**
https://github.com/Vermintide-Mod-Framework/Vermintide-Mod-Framework/issues/21

### CODE REVIEW

File: vmf/scripts/mods/vmf/modules/ui/chat/chat_actions.lua [hunk 1]
```diff
@@ -183,8 +183,24 @@ vmf:hook("ChatGui", "_update_input", function(func, self, input_service, menu_in
 
     -- ctrl + v
     if Keyboard.pressed(Keyboard.button_index("v")) and Keyboard.button(Keyboard.button_index("left ctrl")) == 1 then
-      -- remove CR characters
-      local new_chat_message = self.chat_message .. tostring(Clipboard.get()):gsub(string.char(0x0D), "")
+      local new_chat_message = self.chat_message
+      
+      -- remove carriage returns
+      local clipboard_data = tostring(Clipboard.get()):gsub(string.char(0x0D), "")
```
Comment (FireSiku) [inline vmf/scripts/mods/vmf/modules/ui/chat/chat_actions.lua]: I feel like ```:gsub("\r", "")``` would be a simpler alternative to using string.char

Comment (Aussiemon) [reply vmf/scripts/mods/vmf/modules/ui/chat/chat_actions.lua]: That's part of the original code, I just moved it out of the way and clarified the comment.

I could still make it part of my changes, though.

Comment (FireSiku) [reply vmf/scripts/mods/vmf/modules/ui/chat/chat_actions.lua]: It would sanitize how cluttered that line is a little bit. The rest looks good though.

**Review by FireSiku — APPROVED** [review]

**Comment by Dreomite** [comment]
Yeah, as @FireSiku proposed, `:gsub("\r", "")` would be better. Also, maybe change
```lua
        local valid_data = {}
        clipboard_data:gsub(".", function(c)
          if Utf8.valid(c) then
            table.insert(valid_data, c)
          end
        end)
        new_chat_message = new_chat_message .. table.concat(valid_data)
```
so, when you paste huge texts, lua won't rehash tons of temporarty strings. But it's probably unnessesary optimization and can be skipped.

**Pull request merged**